tests/test_core.py

```diff
@@ -110,11 +110,13 @@ def test_different_schema(simple_table):
 def test_partition_filter(partition_table):
     # partition filter
     df = ddt.read_delta_table(partition_table, version=0, filter=[("col1", "==", 1)])
+    assert df.npartitions == 11
     assert df.compute().shape == (21, 3)
 
     df = ddt.read_delta_table(
         partition_table, filter=[[("col1", "==", 1)], [("col1", "==", 2)]]
     )
+    assert df.npartitions == 21
     assert df.compute().shape == (39, 4)
```

Comment:
Can you please add a test for

- Filter on regular, non-partition column
- Filter on mixed set with at least one partition column and at least one non-partition column

Reply:
I understand that all of this is tested in the `test_partition_filters` cases but I believe there is value to test this on the full API to makes sure we're not encountering any weird errors in conjunction with deltalake or pyarrow

Reply:
👍

Reply:
I added more test cases to `test_partition_filter`. There were already test cases with filtering, but adding more revealed some bugs, so that was good. It's ready for another look.